Sanketika-Obsrv/obsrv-core
# Metrics implementation for MasterDataIndexerJob

Changes -

1. Updated job to work with service account.
2. Job compatible for multiple cloud storage provider.
3. Added metrics for the job.
4. Sample metric event ->
{
"eid": "METRIC",
"ets": 1700474870038,
"mid": "b7e57824-baa9-4eb2-b452-00b1c74abd84",
"actor": {
"id": "MasterDataProcessorIndexerJob",
"type": "SYSTEM"
},
"context": {
"env": "local",
"pdata": {
"id": "DataProducts",
"pid": "MasterDataProcessorIndexerJob",
"ver": "1.0.0"
}
},
"object": {
"id": "telemetry-content-data",
"type": "Dataset",
"ver": "1.0.0"
},
"edata": {
"metric": {
"metric_name": 1,
"total_time_taken": 34977,
"total_events_processed": 10000, ...etc.
},
"labels": [
{
"key": "job",
"value": "MasterDataIndexer"
},
{
"key": "datasetId",
"value": "telemetry-content-data"
},
{
"key": "cloud",
"value": "aws"
}
]
}
}

Test scenarios to be performed -

1. Job should filter dataset based on indexData column and check if there are more than one datasource associated with each dataset. If so, return early.
2. Job should create data file and upload to respective cloud storage provider.
3. Submit ingestion spec to druid for processing.
4. Update datasourceRef if needed.
5. Job should generate metrics for both success and failure scenarios.
